### PR TITLE
refactor: remove ha_get_helper_schema (closes #1186)

### LINKED ISSUE #1186: [FEATURE] Remove ha_get_helper_schema after #1179 auto-attach
## What would you like to see?

With #1179 merged, `ha_config_set_helper` validation errors auto-attach `data_schema` for all 27 helper types, plus a `data_schema_unavailable_reason: "menu_helper_requires_branch"` marker for `template`/`group` without a derivable menu choice. The standalone `ha_get_helper_schema` now duplicates the failure-path schema discovery — call-and-correct returns the same content.

Proposal: remove `ha_get_helper_schema`, consolidate the upfront-discovery path into the `home-assistant-best-practices` skill. Per AGENTS.md "Tool Consolidation": preserving functionality through an alternative path is not a breaking change.

Steps:
1. Add per-type schema reference content (simple-helper field tables + flow-helper schema patterns + menu-rooted two-step flow) to `home-assistant-best-practices/references/helper-selection.md` upstream in `homeassistant-ai/skills`.
2. Cover `template`/`group` menu-options discovery — either:
   - **a.** Include `menu_options` directly in the validation-error context for menu-rooted types (extending the existing `data_schema_unavailable_reason` marker, ~5 LOC).
   - **b.** Rely on skill content alone; LLMs that skip the skill discover via fail-and-retry.
3. Delete `ha_get_helper_schema`, the `ALL_HELPER_TYPES` Literal, and the simple-type dispatch branch in `tools_config_entry_flow.py`.
4. Drop the `ha_get_helper_schema` cross-reference from `ha_config_set_helper`'s docstring.
5. Keep `SIMPLE_HELPER_SCHEMAS` in source — still consumed by the auto-attach on errors.

## Why do you need this?

`ha_get_helper_schema` was a workaround for `ha_config_set_helper` not surfacing schemas inline; #1179 closed that gap. AGENTS.md flags 92+ tools as past the tool-selection-accuracy threshold. The tool's remaining value is speculative pre-reads (rare in practice) and menu-options discovery for two helpers, both addressable without a dedicated tool.

The "LLM creates a degenerate helper without an error" risk is partially covered: any helper with required-field validation surfaces the schema on failure. For helpers that accept anything HA validates (e.g. `input_boolean` with just `name`), failure modes are functional rather than structural — schema attach wouldn't have prevented them either. The skill remains the comprehensive reference for callers that read it; LLMs that skip the skill aren't worse off than today.

## Additional context

- Closes the speculative half of #1149 that #1179 deferred (per its PR body's "Future improvements" section).
- Patch76's #1179 framed tool removal as breaking; per AGENTS.md it isn't, since functionality is preserved through the error path + skill.
- The skills repo is submoduled (`src/ha_mcp/resources/skills-vendor/` → `homeassistant-ai/skills`); step 1 is a separate PR there.
- Step 2a vs 2b is a real design decision worth surfacing in the discussion — 2a removes the skill-bypass risk for the menu-helpers case at the cost of 5 LOC.

### PR DESCRIPTION
**Pull request by kingpanther13**
## What does this PR do?

Removes `ha_get_helper_schema` per #1186. The standalone introspection tool became redundant when #1179 added auto-attached `data_schema` to `ha_config_set_helper` validation errors — agents can self-correct from the first 4xx without a separate discovery round-trip.

For menu-rooted helpers (`template`, `group`) where no sub-type has been picked yet, the validation-error context now also carries `menu_options` inline (step 2a from the issue), so the caller can pick a branch on the next try without a separate discovery round-trip.

Per-type schema reference content lives in the `home-assistant-best-practices` skill — see [homeassistant-ai/skills#47](https://github.com/homeassistant-ai/skills/pull/47).

Per AGENTS.md "Tool Consolidation", removing a tool whose functionality is preserved through an alternative path is **not** a breaking change.

### Removed
- `ha_get_helper_schema` tool + `ConfigEntryFlowTools` class + `ALL_HELPER_TYPES` Literal in `tools_config_entry_flow.py`
- `"Call ha_get_helper_schema"` suggestions from `_raise_flow_api_error`
- `ha_get_helper_schema` cross-references from `ha_config_set_helper` and `ha_config_list_helpers` docstrings + `server.py` lite docstrings

### Added
- `fetch_helper_menu_options()` helper — best-effort menu_options fetcher for use by error-context builders
- `menu_options` in `_flow_helper_error_context` for menu-rooted pre-flow validation errors (issue #1186 step 2a)

### Tests
- Deleted `ha_get_helper_schema` e2e tests (no longer applicable)
- Deleted `TestHaGetHelperSchemaSimpleDispatch` unit test class
- Updated `test_menu_rooted_without_choice_surfaces_unavailable_reason` and `test_template_without_menu_key_surfaces_marker` to assert `menu_options` in context
- Added `TestFetchHelperMenuOptions` covering the new helper (4 cases: menu flow, form flow, HA failure, non-string filter)
- Updated `test_create_group_helper_missing_menu_selection` e2e to assert `menu_options` inline on the error

## Type of change
- [x] 🔧 Maintenance/refactor

## Testing
- [ ] I have tested these changes with a LLM agent
- [ ] All automated tests pass (`uv run pytest`)
- [x] Code follows style guidelines (`uv run ruff check`) — ruff passes locally

## Future improvements

None planned.

## Checklist
- [x] I have updated documentation if needed — docstring references updated; `tools.json`/`README.md`/`DOCS.md` will auto-regen via `sync-tool-docs.yml` on merge.

### CODE REVIEW

File: src/ha_mcp/tools/tools_config_helpers.py [hunk 1]
```diff
@@ -651,6 +651,9 @@ async def _flow_helper_error_context(
         context["data_schema"] = schema
     elif helper_type in _MENU_ROOTED_FLOW_HELPER_TYPES and not menu_choice:
         context["data_schema_unavailable_reason"] = "menu_helper_requires_branch"
+        menu_options = await fetch_helper_menu_options(client, helper_type)
+        if menu_options:
+            context["menu_options"] = menu_options
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply src/ha_mcp/tools/tools_config_helpers.py]: ✅ Addressed in a2aff1eb. Merged `_fetch_data_schema_for_error_context` and `fetch_helper_menu_options` into a single `fetch_helper_flow_info()` that returns `{"schema"?: list, "menu_options"?: list}` from one HA round-trip. Both `_raise_flow_api_error` and `_flow_helper_error_context` now use it. Net: one `start_config_flow` call instead of two on the menu-rooted-no-choice path, and the swallow/breadcrumb pattern is single-sourced.

File: src/ha_mcp/tools/tools_config_entry_flow.py [hunk 2]
```diff
@@ -352,6 +307,47 @@ async def _fetch_data_schema_for_error_context(
...
+        if intro_flow_id:
+            try:
+                await asyncio.wait_for(
+                    client.abort_config_flow(intro_flow_id), timeout=5.0
+                )
+            except Exception as abort_err:
+                logger.debug(
+                    f"Failed to abort introspection flow {intro_flow_id}: {abort_err}"
+                )
+
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply src/ha_mcp/tools/tools_config_entry_flow.py]: ✅ Addressed in a2aff1eb. Same unification — `fetch_helper_menu_options` is gone; `fetch_helper_flow_info` returns both `schema` and `menu_options` from a single flow-start call.

**Review by kingpanther13 — COMMENTED** [review]
✅ Addressed all PR #1315 review feedback in a2aff1eb.

**Gemini (2 inline threads, both about unification):** Merged `_fetch_data_schema_for_error_context` and `fetch_helper_menu_options` into a single `fetch_helper_flow_info()` returning `{"schema"?: list, "menu_options"?: list}` from one HA round-trip. Both `_raise_flow_api_error` and `_flow_helper_error_context` use the unified helper.

**Internal review findings also addressed:**
- Silent-failure-hunter (HIGH/MEDIUM defensive-wrap concerns): incidentally fixed by unification — one call site to wrap (already wrapped) and one outer swallow with breadcrumb.
- Test analyzer (rating 6): added `test_menu_rooted_marker_omits_menu_options_on_ha_failure` covering the marker-set-but-no-menu_options edge case.
- Code reviewer (C2): added `test_menu_options_absent_when_list_is_empty` and `test_menu_options_absent_when_key_missing` to lock the empty/missing contract.
- Code reviewer (C1): the unstructured-branch 4xx in `_raise_flow_api_error` is now sourced from the same introspection result as the structured branch.

**Review by Patch76 — COMMENTED** [review]
The unification into `fetch_helper_flow_info` is a clean win — one HA round-trip per error-context build instead of two, and the `{"schema"?, "menu_options"?}` shape gives both branches of `_raise_flow_api_error` symmetric access without branching on which fetch to call.

Deletion scope is consistent: tool + `ConfigEntryFlowTools` + `ALL_HELPER_TYPES` Literal + all docstring cross-references in one pass, including the suggestion line in `_raise_flow_api_error` and the lite-docstrings test. No orphan references.

`TestFetchHelperFlowInfo` (8 cases) covers form / menu+choice / menu without choice / HA failure / non-string filter / submit failure / "absent ≠ empty list" / "absent ≠ missing key" — the contract surface is well-locked against future regressions. The companion `test_menu_rooted_marker_omits_menu_options_on_ha_failure` pins the negative-side behaviour for `_flow_helper_error_context`.

**Review by Patch76 — APPROVED** [review]
All checks green, contract surface well-locked, unification round-trip-economical — approving.
